### PR TITLE
Variable weights improvements additions

### PR DESCRIPTION
**Pull request by lin-d-hop**
From #568 

Thanks @RohanM for any feedback on this one

xx

### CODE REVIEW

File: app/serializers/api/admin/line_item_serializer.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def units_variant
     Api::Admin::UnitsVariantSerializer.new(object.variant).serializable_hash
   end
 
-  def unit_value
-    object.unit_value.to_f
+  def final_weight_volume
+    object.final_weight_volume.to_f
```
Comment (RohanM) [inline app/serializers/api/admin/line_item_serializer.rb]: How come we're casting to float here rather than leaving as a decimal?

**Comment by RohanM** [comment]
Hi @lin-d-hop,

That's looking good :) I made a minor comment about float vs. decimal usage, but otherwise looking fine. I've pulled this branch into the main repo as uk/variable-weights-improvements so we can get a build running.

I think this might be related the the switch to the new branch, but next time could you divide the work up into smaller commits? It took a while to get my head around what was going on, reading the whole thing as one lump.

But otherwise looking good. If it builds fine I'll pass it on to Sally to give it a test in staging.

File: app/serializers/api/admin/line_item_serializer.rb (hunk 1 above)
Comment (lin-d-hop) [inline app/serializers/api/admin/line_item_serializer.rb]: Yeah not sure. The serializer I believe was changed by @oeoeaio in some bulk management refactoring? So I just kept the cast to float as per his changes.

**Comment by lin-d-hop** [comment]
Thanks @RohanM. Yeah I've realised as well that the big commits are bad practice and confusing. Thanks for the prompt to shrink them down.

Thanks for feedback. @sstead let me know if anything ugly comes up in testing.

xx

**Comment by lin-d-hop** [comment]
@sstead Any news on the testing of this? Thanks xx

**Comment by kirstenalarsen** [comment]
Hi @lin-d-hop, @sstead and I not sure but think that @RohanM didn't get a chance to look at this before he left - Sally wasn't ever asked to test it. Have just added it to @mkllnk's list so will probably have a go at merging it in next week.

**Comment by mkllnk** [comment]
Unfortunately, the tests failed on this branch.
```
spec/features/admin/bulk_order_management_spec.rb:127

     Failure/Error: Unable to find matching line from backtrace
     ActiveRecord::StatementInvalid:
       PG::Error: ERROR:  deadlock detected
       DETAIL:  Process 29899 waits for AccessExclusiveLock on relation 4383665 of database 4383174; blocked by process 30092.
       Process 30092 waits for RowShareLock on relation 4383327 of database 4383174; blocked by process 29899.
       HINT:  See server log for query details.
: ALTER TABLE "cms_categories" DISABLE TRIGGER ALL;...
```

I'm wondering if that is a problem with our test setup. I'll run that test on my machine to see if that's working.

**Comment by mkllnk** [comment]
Okay, I have problems on my computer as well. The bulk order management fails to load any orders. It's just hanging while loading. This needs further investigation.

**Comment by mkllnk** [comment]
Oh, hang on, the hanging bulk order management was because I forgot to migrate the database. I'll run the tests again...

**Comment by mkllnk** [comment]
Still get errors. But this time it looks quite useful:
```
An error occurred in an after hook
  NoMethodError: undefined method `final_weight_volume' for #<Spree::LineItem:0x0000000d10bd98>
  occurred at /home/maikel/.bundler/ofn/ruby/1.9.1/gems/activemodel-3.2.21/lib/active_model/attribute_methods.rb:407:in `method_missing'

  1)                                                                            
  As an Administrator
  I want to be able to manage orders in bulk
 listing orders displaying the list of line items displays a list of line items
     Failure/Error: page.should have_selector "tr#li_#{li1.id}"
     Capybara::ExpectationNotMet:
       expected to find css "tr#li_13" but there were no matches
     # ./spec/features/admin/bulk_order_management_spec.rb:37:in `block (4 levels) in <top (required)>'
     # -e:1:in `<main>'
```

Is that something you can reproduce?

**Comment by mkllnk** [comment]
Me again. I rebuild your branch every time we updated the master and the fourth build passed! So I think this branch should be fine and can be merged. @kirstenalarsen should we stage it?

**Comment by mkllnk** [comment]
It's staged on https://staging2.openfood.com.au/.

File: app/serializers/api/admin/line_item_serializer.rb (hunk 1 above)
Comment (mkllnk) [inline app/serializers/api/admin/line_item_serializer.rb]: Actually, unit_value can be `nil` and `nil.to_f` is `0.0`. Maybe that was the idea behind.

**Pull request merged**